p2p/integration_test.go

```diff
@@ -0,0 +1,138 @@
+package p2p
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/spacemeshos/go-spacemesh/p2p/service"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/sync/errgroup"
+	"math/rand"
+	"sync/atomic"
+	"testing"
+	"time"
+)
+
+type integrationTest struct {
+	networkSize       int
+	bootNodes         int
+	randomConnections int
+}
+
+func (st *integrationTest) Run(t testing.TB, testFunc func(t testing.TB, network *P2PSwarm)) {
+	fmt.Println("===============================================================================================")
+	fmt.Println("Running a test with these parameters :" + spew.Sdump(st))
+	tim := time.Now()
+	network := NewP2PSwarm(nil, nil)
+	network.Start(t, st.bootNodes, st.networkSize, st.randomConnections)
```

Comment:
this can be done using testify test suites...

Reply:
#372